### PR TITLE
don't hide forum topics if last post is from an unverified user

### PR DESCRIPTION
**Pull request by Jamiras**
Instead of filtering on forum topics where the last post is unauthorized, this modifies the query to count the number of authorized posts in a forum topic and ignores anything without any authorized posts.

The list will still show the unauthorized user as the user for the last post, but upon opening the topic, the unauthorized posts will remain hidden.
image

The unauthorized links page still shows the full post count:
image

The unauthorized post also does not appear in the Forum Activity list on the main page, or the Forum Post History. No change was necessary for that behavior.

### CODE REVIEW

File: public/viewtopic.php [hunk 1]
```diff
@@ -252,7 +252,7 @@
 
             if ($nextCommentAuthorised == 0) {
                 // Allow, only if this is MY comment (disclaimer: unofficial), or if I'm admin (disclaimer: unofficial, verify user?)
```
Comment (luchaos) [inline public/viewtopic.php:254]: This comment doesn't seem to align with the implementation, doesn't it?

Comment (Jamiras) [reply public/viewtopic.php:254]: I'm not sure what the "disclaimer: unofficial" means, but the "MY comment" aligns with line 259, and the "I'm admin" aligns with line 255 (after my change).

Comment (Jamiras) [reply public/viewtopic.php:254]: I'm guessing it's supposed to be "disclaimer: unverified" to indicate the disclaimer will be shown.

Comment (luchaos) [reply public/viewtopic.php:254]: Ah, yes - I see it now, my bad

**Review by luchaos — APPROVED** [review]

**Pull request merged**